lib/chai/core/assertions.js

```diff
@@ -1763,10 +1763,22 @@ module.exports = function (chai, _) {
       , isOwn = flag(this, 'own')
       , flagMsg = flag(this, 'message')
       , obj = flag(this, 'object')
-      , ssfi = flag(this, 'ssfi');
+      , ssfi = flag(this, 'ssfi')
+      , nameType = typeof name;
+
+    flagMsg = flagMsg ? flagMsg + ': ' : '';
+
+    if (isNested) {
+      if (nameType !== 'string') {
+        throw new AssertionError(flagMsg + 'the argument to property must be a string when using nested syntax');
+      }
+    } else {
+      if (nameType !== 'string' && nameType !== 'number' && nameType !== 'symbol') {
+        throw new AssertionError(flagMsg + ' the argument to property must be a string, number, or symbol');
```

Comment:
It looks like this has an extra `space`. `flagMsg` variable already has a trailing space